relayer/relays/beacon/header/syncer/syncer.go

```diff
@@ -493,55 +523,127 @@ func (s *Syncer) unmarshalBeaconState(slot uint64, data []byte) (state.BeaconSta
 	return beaconState, nil
 }
 
-// Sanity check the finalized and attested header are at 32 boundary blocks so we can download the beacon state
-func (s *Syncer) findAttestedAndFinalizedHeadersAtBoundary(initialSlot, lowestSlot uint64) (uint64, error) {
-	var headers []uint64
+// Sanity check the finalized and attested header are at 32 boundary blocks, so we can download the beacon state
+func (s *Syncer) FindLatestAttestedHeadersAtInterval(initialSlot, lowestSlot uint64) (uint64, error) {
```

Comment:
Notice there is also another `FindOldestAttestedHeaderAtInterval`, can they be merged or only need 1 of them as semantically `FindOldest(start,upperBound)==FindLatest(upperBound,start)`?